### DISCUSSION TITLE
Exceptions do not print in mounted apps

### DISCUSSION THREAD

**Discussion by camilla007** [fastapi/fastapi#11417] [Questions] [answered]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
main.py 
from fastapi import (
        Request,
        Depends,
        FastAPI,
        WebSocket,
        Cookie,
        Query,
        status,
        Header,
        WebSocket,
        Form,
        )
from fastapi.encoders import jsonable_encoder
from typing import Optional, Union

from fastapi_utils.tasks import repeat_every
from fastapi.responses import FileResponse, HTMLResponse
from fastapi_sqlalchemy import db
from fastapi.openapi.utils import get_openapi
from fastapi.encoders import jsonable_encoder
from fastapi import (
        Request,
        Depends,
        FastAPI,
        WebSocket,
        Cookie,
        Query,
        status,
        Header,
        WebSocket,
        Form,
        )
from fastapi.staticfiles import StaticFiles

from modules.cdv import app as cdv

appname = os.getenv("APPNAME", "CrossCore")
app = FastAPI(title=appname, redoc_url=None, openapi_url="/openapi.json")
app.state.limiter = limiter
app.openapi = main_custom_openapi

app.mount("/cdv", cdv)





cdv.py
from fastapi import FastAPI, Depends, Request
from fastapi.encoders import jsonable_encoder

from fastapi_versioning import VersionedFastAPI, version

from validators.base_models import *
from validators.response_utils import *

app = FastAPI(title="CrossCore CDV", root_path="/cdv", version="0.0.1")
app.state.limiter = limiter

@app.get(
        "/get_banks",
        response_model=ResponseValidatorMulti,
        dependencies=[Depends(total_users)],
        summary="Use this endpoint to retrieve a list of bank names in the CrossCore ecosystem.",
        deprecated=False,
        )
@limiter.limit(DEFAULT_LIMIT_VALUE)
async def get_banks(request: Request):
    """
    Use this endpoint to retrieve a list of bank names in the CrossCore ecosystem. <br><br>
    This endpoint does not have any parameters.
    """
    raise Exception('test')
    #return success_response_single('')
```

### Description

**I cannot see any errors in apps which are mounted onto the main application.**

I run the server with `uvicorn main:app --reload` 

- when I navigate to localhost:8000/docs and run a method that contains a raise Exception('test') it raises the exception and prints the full traceback in the terminal output
- when I navigate to localhost:8000/cdv/docs and run a method that contains a raise Exception('test') I get a 500 internal server error and no traceback printed in the terminal output

How can I get the sub applications that have been mounted to show the traceback? 

I have already tried
- Adding them as APIRouter instead of app.mount(
- Adding custom exception_handlers
- Adding a custom logging.json config to the uvicorn start command
- Investigating the uvicorn/protocols/http site packages as well as others

### Operating System

Linux

### Operating System Details

Linux camasus 6.1.84-1-MANJARO #1 SMP PREEMPT_DYNAMIC Thu Apr  4 04:30:51 UTC 2024 x86_64 GNU/Linux

### FastAPI Version

0.110.0

### Pydantic Version

1.10.14

### Python Version

Python 3.11.8

**Comment by JavierSanchezCastro** [comment]
Hello, can you provide a MRE without redundant code?

**Comment by YuriiMotov** [comment, marked as answer]
I created a simple example and it works fine. I can see tracebacks in both apps:

main.py
```
from fastapi import FastAPI

from cdv import subapi

app = FastAPI()

@app.get("/app")
def read_main():
    raise Exception("exception")

app.mount("/subapi", subapi)
```

cdv.py
```
from fastapi import FastAPI

subapi = FastAPI()

@subapi.get("/sub")
def read_sub():
    raise Exception("sub exception")
```

**Comment by camilla007** [comment]
Thanks to https://github.com/YuriiMotov - your MRE did allow me to see my mistake. It was an issue with how the SQL session was initialized causing a circular import error and has since been resolved.
